### PR TITLE
Operator cancel/retry, and re-gate live games against today's engine

### PR DESCRIPTION
**Pull request by gtanczyk**
Two features, built in parallel as requested: the console's missing verbs, and the break-and-nudge loop that replaces engine pinning.

## Cancel and retry

The console could watch a stuck build and publish a finished one, but the most common alert — a stalled job — had no verb: killing or restarting still meant curl.

**Cancel** transitions the job to `canceled` and is honest about what that does. The Copilot tasks API has no kill endpoint, so cancellation is cooperative: the job being terminal *is* the mechanism — the build channel's stop check now includes the canceled state (`copilot-backend.cancel` has described that behaviour since it was written; nothing implemented it until now), so a live session reads `control.stop` on its next report and winds down, and its writes stop landing. The response says `stopEnforced: false` and the panel renders "canceled — the session stops at its next report" rather than "stopped". Two named refusals: a finished job stays finished, and a job mid-publish is left to land or fail. The button takes two clicks, because canceled has no exit in the transition table.

**Retry** starts a fresh session through the same resume path creator feedback uses — restoring delivered work from the store, or handing an undelivered round its old branch, the only copy of its work. Offered from dead rounds (`failed`, `needs_changes`) and visibly stuck ones (stalled `building`/`dispatched`), not from every healthy build. Honest edges: a job that never reached an agent is refused (`never_dispatched` — no spec on record, no branch to resume, an empty session would spend a credit briefing an agent with nothing); a resume the backend refused answers **502** rather than pretending it took; a same-state retry writes its own `operator_retry` transition so the round a person explicitly started is visible in the history. The ledger books the credit, and the response says so.

## Break-and-nudge (phase 1)

The engine question, answered by regeneration rather than pinning: detect breakage, nudge the creator, make the fix an improvement round.

The premise that makes this safe: **serving is already immune** — a published game is a baked, self-contained bundle. What drifts is whether the game would still *pass* a rebuild, and nothing measured that.

- **Health is a second verdict, never a rewrite of the first.** A re-gate runs the same check on the same stored sources with the engine pin overridden to `main`, recorded as `manifest.health`. The acceptance verdict is provenance — a red re-run erasing it would erase the justification for publishing.
- **Verdicts now name their target.** The gate resolves the commit it actually checked against (`git rev-parse HEAD` in the harness) and the first run stamps it onto the manifest — closing the runner's "closing it is the next change to this file" note from the validation side.
- **The loop:** operator hits *Re-gate* on the console's new published shelf → the same configured Cloud Build trigger runs in `--health` mode (tagged `health`, cost booked to the job that built the game) → the verdict lands on the manifest → the sweep reads it back. Green resolves quietly. **Red nudges the creator** (in-app + email, deep-linked to the studio where the improvement round starts, written as an invitation: "it still plays fine — a quick improvement round would bring it up to date") **and copies the operator**, idempotently.

Deliberately out of this slice, named so it stays visible: nothing auto-re-gates on an engine push yet — the trigger is the operator's button; wiring the games repo's push webhook to sweep the shelf is the follow-up.

## Testing

Full gate green: lint, type-check, 2129 tests, build. New coverage: channel stop on canceled, both routes' refusals and edge cases, two-click cancel, retry offer rules, engine-commit resolution and override, health-verdict-beside-gate-verdict, first-run-pins/never-repins, `--health` flag and tag, regate route, sweep resolution (red/green/stale), nudge idempotency, the published shelf.

---
_Generated by [Claude Code](https://claude.ai/code/session_01P1rDAP357447cbEmwhCXY9)_

### CODE REVIEW

File: apps/web/src/AdminJobsPanel.tsx [hunk 1]
```diff
@@ -37,6 +45,162 @@ const STALL_COPY: Record<NonNullable<JobQueueEntry['stall']>, string> = {
...
+ */
+function healthLabel(game: PublishedGame, now: number): string {
+  const check = game.healthCheck;
+  if (!check) return 'never checked';
+  if (!check.verdictAt) {
+    const minutes = Math.max(0, Math.floor((now - Date.parse(check.requestedAt)) / 60_000));
+    return `checking… (${minutes < 60 ? `${minutes}m` : `${Math.floor(minutes / 60)}h`} ago)`;
+  }
+  return check.green ? `healthy (${check.verdictAt.slice(0, 10)})` : `FAILING (${check.verdictAt.slice(0, 10)})`;
+}
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/web/src/AdminJobsPanel.tsx:96]: Real bug — fixed in 142b442. Both branches are explicit now, and a resolved check missing its boolean reads "verdict unreadable" rather than borrowing the scariest label available. The label and the row highlight can no longer disagree about the same record. No writer produces such a record today, but the type allows it, so the renderer handles it.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/api/src/email-templates.ts [hunk 2]
```diff
@@ -141,6 +141,22 @@ const notificationCopy: Record<
...
+  // improvement round is what brings the game along. Framed as an invitation because
+  // it is one: the creator touching their game again is the good outcome here.
+  'submission.game_health': {
+    en: {
+      subject: 'Your game could use a refresh',
+      lead: 'no longer passes our checks on the latest engine. It still plays fine — but a quick improvement round would bring it up to date.',
+      cta: 'Improve it',
+    },
+    pl: {
+      subject: 'Twoja gra przydałaby się odświeżyć',
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/email-templates.ts]: You're right that it was ungrammatical, but the suggested phrasing has the same problem from the other side: "przydałaby się do odświeżenia" says the game is useful *for* refreshing something. The idiomatic construction is dative — "Twojej grze przydałoby się odświeżenie" ("your game could use a refresh"). Fixed that way in 142b442, in both the email subject and the in-app title (they share the copy). Covers the pl.json thread too.

---
_Generated by [Claude Code](https://claude.ai/code)_
